pkg/proxy/config/config.go

```diff
@@ -69,203 +49,206 @@ type EndpointsConfigHandler interface {
 // EndpointsConfig tracks a set of endpoints configurations.
 // It accepts "set", "add" and "remove" operations of endpoints via channels, and invokes registered handlers on change.
 type EndpointsConfig struct {
-	mux     *config.Mux
-	bcaster *config.Broadcaster
-	store   *endpointsStore
+	informer cache.Controller
+	lister   listers.EndpointsLister
+	handlers []EndpointsConfigHandler
+	// updates channel is used to trigger registered handlers.
+	updates chan struct{}
 }
 
 // NewEndpointsConfig creates a new EndpointsConfig.
-// It immediately runs the created EndpointsConfig.
-func NewEndpointsConfig() *EndpointsConfig {
-	// The updates channel is used to send interrupts to the Endpoints handler.
-	// It's buffered because we never want to block for as long as there is a
-	// pending interrupt, but don't want to drop them if the handler is doing
-	// work.
-	updates := make(chan struct{}, 1)
-	store := &endpointsStore{updates: updates, endpoints: make(map[string]map[types.NamespacedName]*api.Endpoints)}
-	mux := config.NewMux(store)
-	bcaster := config.NewBroadcaster()
-	go watchForUpdates(bcaster, store, updates)
-	return &EndpointsConfig{mux, bcaster, store}
+func NewEndpointsConfig(c cache.Getter, period time.Duration) *EndpointsConfig {
```

Comment:
@wojtek-t could you please modify this so that it takes in an [EndpointsInformer](https://github.com/kubernetes/kubernetes/blob/51beb16edec3173a098d8bb608aa0d4e895cd672/pkg/client/informers/informers_generated/externalversions/core/v1/endpoints.go#L35). In OpenShift, we need to be able to share the caches for endpoints and services between the proxy and dns code (which we run in the same process). We'd need a similar change to `NewServiceConfig` below.

cc @smarterclayton @sttts @deads2k